app/src/main/java/nerd/tuxmobil/fahrplan/congress/repositories/AppRepository.kt

```diff
@@ -276,6 +276,12 @@ object AppRepository {
                 }
             }
         }
+
+        val alarmEventIds = readAlarms().map { it.eventId }.toSet()
```

Comment:
Why do you transform the list to a set? Do you expect duplicates?

Reply:
So lookups in the loop can be done in O(1) and setting the alarms is done in O(n) instead of the O(n^2) variant with nested for loops.